src/esrs/state.rs

```diff
@@ -1,17 +1,43 @@
 use uuid::Uuid;
 
+use crate::esrs::store::EventStoreLockGuard;
 use crate::types::SequenceNumber;
 
 /// The internal state for an Aggregate.
-/// It contains and id representing the aggregate id, an incremental sequence number and a state
-/// defined by the user of this library.
-#[derive(Clone, Debug)]
+/// It contains:
+/// - an id uniquely representing the aggregate,
+/// - an incremental sequence number,
+/// - a lock representing the atomicity of the access to the aggregate,
+/// - a state defined by the user of this library.
 pub struct AggregateState<S> {
     pub(crate) id: Uuid,
     pub(crate) sequence_number: SequenceNumber,
+    pub(crate) lock: Option<EventStoreLockGuard>,
     pub(crate) inner: S,
 }
 
+impl<S: Clone> Clone for AggregateState<S> {
+    fn clone(&self) -> Self {
+        Self {
+            id: self.id,
+            sequence_number: self.sequence_number,
+            inner: self.inner.clone(),
+            lock: None,
+        }
+    }
+}
```

Comment:
Manual implementation of `Clone`, as it could not be automatically derived anymore.
I am not sure `AggregateState` should even be Clone to be fair - I actually think we should remove this since:
1. I defaulted lock to None, but we probably should think carefully about this choice;
2. regardless, you can individually clone the "data transfer" fields if you need. The AggregateState type should only be used to thread from `Manager::load` to `Manager::handle`, and cloning between these two operations does not make sense (handling the same state again would incur into sequence number locking!).